homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -147,80 +148,84 @@ def __init__(
         self._time = departuretime
-        self._state: TrainStop | None = None
         self._attr_extra_state_attributes = {}
 
     async def async_update(self) -> None:
         """Retrieve latest state."""
         when = datetime.now()
+        _state: TrainStop | None = None
         if self._time:
             departure_day = next_departuredate(self._weekday)
             when = datetime.combine(departure_day, self._time).astimezone(TIMEZONE)
         try:
             if self._time:
-                self._state = await self._train_api.async_get_train_stop(
+                _state = await self._train_api.async_get_train_stop(
                     self._from_station, self._to_station, when
                 )
             else:
 
-                self._state = await self._train_api.async_get_next_train_stop(
+                _state = await self._train_api.async_get_next_train_stop(
                     self._from_station, self._to_station, when
                 )
         except ValueError as output_error:
             _LOGGER.error("Departure %s encountered a problem: %s", when, output_error)
 
-        if not self._state:
+        if not _state:
             self._attr_available = False
             self._attr_native_value = None
             self._attr_extra_state_attributes = {}
             return
 
         self._attr_available = True
 
-        self._attr_native_value = self._state.advertised_time_at_location.astimezone(
+        self._attr_native_value = _state.advertised_time_at_location.astimezone(
             TIMEZONE
         )
-        if self._state.time_at_location:
-            self._attr_native_value = self._state.time_at_location.astimezone(TIMEZONE)
-        if self._state.estimated_time_at_location:
-            self._attr_native_value = self._state.estimated_time_at_location.astimezone(
+        if _state.time_at_location:
+            self._attr_native_value = _state.time_at_location.astimezone(TIMEZONE)
+        if _state.estimated_time_at_location:
+            self._attr_native_value = _state.estimated_time_at_location.astimezone(
                 TIMEZONE
             )
 
-        self._attr_extra_state_attributes[ATTR_DEPARTURE_STATE] = (
-            self._state.get_state().name if self._state.get_state().name else None
-        )
-        self._attr_extra_state_attributes[ATTR_CANCELED] = (
-            self._state.canceled if self._state.canceled else None
-        )
-        delay_in_minutes = self._state.get_delay_time()
-        self._attr_extra_state_attributes[ATTR_DELAY_TIME] = (
-            delay_in_minutes.total_seconds() / 60 if delay_in_minutes else None
-        )
-        self._attr_extra_state_attributes[ATTR_PLANNED_TIME] = (
-            as_utc(
-                self._state.advertised_time_at_location.astimezone(TIMEZONE)
+        self._attr_extra_state_attributes = await self.async_update_attributes(_state)
+
+    async def async_update_attributes(self, state: TrainStop) -> dict[str, Any]:
```

Comment:
No need to make this async, and no need to return a value:
```suggestion
    def _update_attributes(self, state: TrainStop) -> None:
```